### PR TITLE
Extract environment setup for the CI into separate action

### PR DESCRIPTION
**Pull request by matamegger**
## Description

There is a lot of redundancy in the CI workflows regarding the setup of the tooling. Which is especially noticeable when updating or changing tooling setups and versions.

## Changes

Extract the setup into a configurable action and replace the setup with the newly introduced action.

## Checklist
- [x] 🗒 `CHANGELOG` entry

### CODE REVIEW

File: .github/actions/setup-build/action.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        registry-url: 'https://registry.npmjs.org/'
+        cache: 'yarn'
+
+    - name: Install node_modules
+      if: ${{ inputs.node == 'true' || inputs.subprojects == 'true' }}
+      shell: bash
+      run: yarn install --frozen-lockfile
+
+    - name: Install node_modules (example/)
+      if: ${{ inputs.subprojects == 'true' || inputs.subprojects == 'true' }}
```
Comment (rolandkakonyi) [inline .github/actions/setup-build/action.yml:56]: Is this a typo? `inputs.subprojects == 'true'` is there 2x!

File: .github/actions/setup-build/action.yml
Comment (rolandkakonyi) [inline .github/actions/setup-build/action.yml:1]: Could we call this `prepare-dependencies`?
`setup-build` is a bit unclear to me if it actually builds or not.

Comment (matamegger) [reply .github/actions/setup-build/action.yml:1]: "dependencies" is hear a bit broadly used IMO. Installing xcode or java is strictly preparing a dependency, but not what we usually talk about when talking about dependencies I think.

`prepare-environment` or `setup-environment`?

Comment (rolandkakonyi) [reply .github/actions/setup-build/action.yml:1]: It is a dependency for the upcoming build process. We use the term in the iOS SDK, but I like both `prepare-environment` and `setup-environment`. Either is fine!

File: .github/actions/setup-build/action.yml (hunk 1 above)
Comment (matamegger) [reply .github/actions/setup-build/action.yml:56]: Embarrassing. Seems to be a copy past error on my side. should be just once there.

File: .github/actions/setup-build/action.yml
Comment (matamegger) [reply .github/actions/setup-build/action.yml:1]: I see. I guess `environment` is a good compromise then :)

Comment (matamegger) [reply .github/actions/setup-build/action.yml:1]: https://github.com/bitmovin/bitmovin-player-react-native/pull/408/commits/e25bdbf939ee04d175fbeea0abe4c22b680f4a90

File: .github/actions/setup-build/action.yml (hunk 1 above)
Comment (matamegger) [reply .github/actions/setup-build/action.yml]: https://github.com/bitmovin/bitmovin-player-react-native/pull/408/commits/443a4be8423bfb1c314ceef4760ff22aa6502365

**Pull request merged**